File: apps/ios/Centwise/Core/Design/Theme/CentwiseTypography.swift
import SwiftUI

public enum CentwiseTypography {
    public static let largeTitle = Font.system(size: 30, weight: .bold, design: .rounded)
    public static let title1 = Font.system(size: 24, weight: .bold, design: .rounded)
    public static let title2 = Font.system(size: 20, weight: .bold, design: .rounded)
    public static let title3 = Font.system(size: 16, weight: .semibold, design: .rounded)
    public static let headline = Font.system(size: 15, weight: .semibold, design: .default)
    public static let body = Font.system(size: 14, weight: .regular, design: .default)
    public static let bodyMedium = Font.system(size: 14, weight: .medium, design: .default)
    public static let subheadline = Font.system(size: 13, weight: .regular, design: .default)
    public static let footnote = Font.system(size: 12, weight: .regular, design: .default)
    public static let caption = Font.system(size: 11, weight: .medium, design: .default)
    public static let caption1 = Font.system(size: 11, weight: .medium, design: .default)
    public static let caption2 = Font.system(size: 10, weight: .regular, design: .default)

    // MARK: - Amount & Numerical Typography (Monospaced digits)
    public static let amountHero = Font.system(size: 26, weight: .bold, design: .rounded).monospacedDigit()
    public static let amountLarge = Font.system(size: 21, weight: .bold, design: .rounded).monospacedDigit()
    public static let amountMedium = Font.system(size: 15, weight: .semibold, design: .rounded).monospacedDigit()
    public static let amountSmall = Font.system(size: 13, weight: .medium, design: .rounded).monospacedDigit()
}

File: apps/android/app/src/main/kotlin/com/centwise/core/design/theme/CentwiseTypography.kt
package com.centwise.core.design.theme

import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.sp

/**
 * Idiomatic Jetpack Compose Typography system aligned 1:1 with iOS CentwiseTypography.swift.
 * Ensures consistent, clean, and responsive sizing across all screen densities without text overflow.
 */
object CentwiseTypography {
    // MARK: - Standard Text Styles (Matching iOS Font.system)
    val LargeTitle = TextStyle(
        fontFamily = FontFamily.Default,
        fontWeight = FontWeight.Bold,
        fontSize = 30.sp,
        letterSpacing = (-0.5).sp
    )

    val Title1 = TextStyle(
        fontFamily = FontFamily.Default,
        fontWeight = FontWeight.Bold,
        fontSize = 24.sp,
        letterSpacing = (-0.3).sp
    )

    val Title2 = TextStyle(
        fontFamily = FontFamily.Default,
        fontWeight = FontWeight.Bold,
        fontSize = 20.sp
    )

    val Title3 = TextStyle(
        fontFamily = FontFamily.Default,
        fontWeight = FontWeight.SemiBold,
        fontSize = 16.sp
    )

    val Headline = TextStyle(
        fontFamily = FontFamily.Default,
        fontWeight = FontWeight.SemiBold,
        fontSize = 15.sp
    )

    val Body = TextStyle(
        fontFamily = FontFamily.Default,
        fontWeight = FontWeight.Normal,
        fontSize = 14.sp
    )

    val BodyMedium = TextStyle(
        fontFamily = FontFamily.Default,
        fontWeight = FontWeight.Medium,
        fontSize = 14.sp
    )

    val Subheadline = TextStyle(
        fontFamily = FontFamily.Default,
        fontWeight = FontWeight.Normal,
        fontSize = 13.sp
    )

    val Footnote = TextStyle(
        fontFamily = FontFamily.Default,
        fontWeight = FontWeight.Normal,
        fontSize = 12.sp
    )

    val Caption = TextStyle(
        fontFamily = FontFamily.Default,
        fontWeight = FontWeight.Medium,
        fontSize = 11.sp
    )

    val Caption1 = TextStyle(
        fontFamily = FontFamily.Default,
        fontWeight = FontWeight.Medium,
        fontSize = 11.sp
    )

    val Caption2 = TextStyle(
        fontFamily = FontFamily.Default,
        fontWeight = FontWeight.Normal,
        fontSize = 10.sp
    )

    // MARK: - Numerical & Currency Amounts (Matching iOS monospacedDigit)
    val HeroAmount = TextStyle(
        fontFamily = FontFamily.Default,
        fontWeight = FontWeight.Bold,
        fontSize = 26.sp,
        letterSpacing = (-0.4).sp
    )

    val AmountLarge = TextStyle(
        fontFamily = FontFamily.Default,
        fontWeight = FontWeight.Bold,
        fontSize = 21.sp
    )

    val AmountMedium = TextStyle(
        fontFamily = FontFamily.Default,
        fontWeight = FontWeight.SemiBold,
        fontSize = 15.sp
    )

    val AmountSmall = TextStyle(
        fontFamily = FontFamily.Default,
        fontWeight = FontWeight.Medium,
        fontSize = 13.sp
    )
}
